README.horizontal-scaling.md

```diff
@@ -0,0 +1,160 @@
+automatically with the necessary credentials and other environment variables from 
+your cluster config settings. If you change any of the scaling
+configuration settings (detailed below) you should re-rerun this recipe.
+
+### Scaling Strategies
+
+There are two high-level strategies under which **moscaler** can be run: `time` and `auto`.
+The time-based strategy ensures that a fixed number of workers are online based on
+the time of day and the day of the week. The `auto` strategy tries to turn on/off 
+worker nodes based on the current state of the cluster, e.g., a server load metric or
+job count values reported by Matterhorn.
+
+The stragegy to be used is determined by the `moscaler_strategy` value in the `moscaler`
+block of your cluster config. 
+
+### Time-based
+
+The number of worker instances available at any given time is controlled by
+three values that can be modified in your `custom_json`.
+
+Example config:
+
+    {
+      "stack": {
+        "chef": {
+          "custom_json": {
+            "moscaler": {
+              "moscaler_strategy": "time",
+              "offpeak_instances": 2,
+              "peak_instances": 10,
+              "weekend_instances": 1
+            },
+            ...
+          }
+        }
+      }
+    }
+
+The (currently non-configurable) time ranges for *offpeak*, *peak* and *weekend* are:
+
+* peak: M-F, 7am-11pm
+* offpeack: M-F, 11pm-7am
+* weekend: 12am Sat - 12pm Sun
+
+### Auto-scaling
+
+There are two types of auto scaling mechanisms available: `LayerLoad` and `HighLoadJobs`.
+Only the `LayerLoad` type should be used at this time as the `HighLoadJobs` mechanism
+makes use of some unreliable data from the Matterhorn statistics API.
+
+#### LayerLoad
+
+This method scales workers up/down based on the value of an Opsworks
+layer's cloudwatch metric. It compares the metric's reported values against a couple
+of threshold settings and scales up or down accordingly.
+
+Example config:
+
+    {
+      "stack": {
+        "chef": {
+          "custom_json": {
+            "moscaler": {
+              "moscaler_strategy": "auto",
+              "autoscale_type": "LayerLoad"
+              "autoscale_up_increment" => 2,
+              "autoscale_down_increment" => 2,
+              "autoscale_up_threshold" => 6.0,
+              "autoscale_down_threshold" => 10.0,
+              "autoscale_cloudwatch_metric" => "load_5"
+            },
+            ...
+          }
+        }
+      }
+    }
+    
+This configuration would result in `moscaler` scaling up two workers if the `load_5` metric
+for the workers layer reports a value equal to or higher than 6.0 at 1 minute intervals
+for 5 consecutive minutes. It would attempt to scale down two workers if that same metric 
+was less than 10.0 for 5 consectutive minutes. (Note: the 5min sample size and 1min intervals 
+are not currently configurable but could be made so if needed.)
+
```

Comment:
I updated the example to be less counter-intuitive, but to answer the question, sort of. There is a difference between the scale up/down conditions being met and actual workers being turned on/off. I added a short section to the readme to try and explain this:

**Question**: what happens if `scale_down_threshold` is actually greater than `scale_up_threshold`? For instance, the
cluster should scale up when the worker layer's `load_5` value is greater than 8.0, but scale down when the value is
less than 10.0.

**Answer**: This may seem counter-intuitive, but it's most likely fine. Keep in mind that `mo-scaler` has
protections built in to prevent workers from being scaled down if they are not idle or if they haven't used up enough
of their billing hour. *A scale down event triggered by the autoscaling only results in an **attempt** to scale down.`
The attempt will frequently be aborted if there's not enough idle workers. Crossing that `autoscale_down_threshold`
just tells `mo-scaler` to start *trying* to scale down.